### PR TITLE
Indexer event query e2e

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

https://mysten.atlassian.net/browse/SUI-1740

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by wlmyng** [comment]
Originally sent to Ge but posting for posterity

I originally thought to
1. Create NFT
2. Transfer NFT
and check the corresponding events. But for some reason I can't seem to be able to find the transfer object event. I'm pretty sure it should emit an event though, at least per the docs at https://docs.sui.io/devnet/build/event_api. So for now I've modified it to:
1. Create NFT
2. Create NFT
3. Transfer coin
The test checks for and expects only the 2 emitted Create NFT events, and then checks that Transfer NFT correctly transfers to recipient address.

Secondly, is there any particular reason why we don't include ClientCommands in the Cargo.toml? Was going to do something like ClientCommands::CreateExampleNFT but noticed that it's not a dependency on sui-indexer. It would make testing a bit more flexible as I think the util functions from test_utils::transaction are quite prescriptive.

File: crates/sui-indexer/tests/integration_tests.rs [hunk 1]
```diff
@@ -168,6 +175,76 @@ mod pg_integration {
...
+        wait_until_transaction_synced(&store, digest_two.base58_encode().as_str()).await;
+        let (transferred_object, sender, receiver, digest_three, _, _) = transfer_coin(context).await.unwrap();
+        wait_until_transaction_synced(&store, digest_three.base58_encode().as_str()).await;
+
+        // Test various ways of querying events
+        let to_query = EventFilter::Sender(sender);
+        let query_response = indexer_rpc_client
+            .query_events(to_query, None, None, None)
+            .await?;
+        // Interestingly, transferring coins do not emit events?
```
[the remark this answers is not in the document]
Comment (wlmyng) [reply crates/sui-indexer/tests/integration_tests.rs:196]: Oh gotcha I see. Would you say that this page is outdated https://docs.sui.io/build/event_api? If "A Sui node emits the following types of events:

[Move event](https://docs.sui.io/build/event_api#move-event)
[Publish event](https://docs.sui.io/build/event_api#publish-event)
[Transfer object event](https://docs.sui.io/build/event_api#transfer-object-event)
[Delete object event](https://docs.sui.io/build/event_api#delete-object-event)
[New object event](https://docs.sui.io/build/event_api#new-object-event)
[Epoch change event](https://docs.sui.io/build/event_api#epoch-change-event)" then I'd assume indexer would track these too

File: crates/sui-indexer/tests/integration_tests.rs [hunk 2]
```diff
@@ -168,6 +175,76 @@ mod pg_integration {
...
+        wait_until_transaction_synced(&store, digest_one.base58_encode().as_str()).await;
+        let (_, _, digest_two) = create_devnet_nft(context).await.unwrap();
+        wait_until_transaction_synced(&store, digest_two.base58_encode().as_str()).await;
+        let (transferred_object, sender, receiver, digest_three, _, _) = transfer_coin(context).await.unwrap();
+        wait_until_transaction_synced(&store, digest_three.base58_encode().as_str()).await;
+
+        // Test various ways of querying events
+        let to_query = EventFilter::Sender(sender);
+        let query_response = indexer_rpc_client
+            .query_events(to_query, None, None, None)
```
Comment (gegaowp) [inline crates/sui-indexer/tests/integration_tests.rs:194]: nit: from_query?

File: crates/sui-indexer/tests/integration_tests.rs [hunk 3]
```diff
@@ -168,6 +175,76 @@ mod pg_integration {
...
+            .await?;
+        // Interestingly, transferring coins do not emit events?
+        assert_eq!(query_response.data.len(), 2);
+        assert_eq!(query_response.data[0].transaction_module, ident_str!("devnet_nft").into());
```
Comment (gegaowp) [inline crates/sui-indexer/tests/integration_tests.rs:198]: can we also assert the package & function here?

File: crates/sui-indexer/tests/integration_tests.rs [hunk 4]
```diff
@@ -168,6 +175,76 @@ mod pg_integration {
...
+            Some(SuiObjectDataOptions::new().with_type()),
+            None,
+            None,
+            None,
+        )
+        .await?
+        .data
+        .into_iter()
+        .filter_map(|object_resp| match object_resp {
+            SuiObjectResponse::Exists(obj_data) => Some(obj_data),
```
Comment (gegaowp) [inline crates/sui-indexer/tests/integration_tests.rs:238]: this is great

**Review by gegaowp — APPROVED** [review]
looks great, just a couple of minor comments.

**Pull request merged**